### PR TITLE
expose store_offset and query_offset in consumer

### PR DESCRIPTION
**Pull request by DanielePalaia**
This closes https://github.com/rabbitmq/rabbitmq-stream-rust-client/issues/195

Exposes store_offset and query_offset in Consumer struct

### CODE REVIEW

**Review by Gsantomaggio — CHANGES_REQUESTED** [review]
can you add the `Offset Not Found Error` test case? for the query offset in case it does not exist.

**Comment by Gsantomaggio** [comment]
Can you add a test using a real offset coming form the chunk ?

**Comment by DanielePalaia** [comment]
Done!

**Review by Gsantomaggio — APPROVED** [review]

File: src/consumer.rs [hunk 1]
```diff
@@ -137,6 +149,44 @@ impl Consumer {
...
+                self.internal.stream.as_str(),
+                offset,
+            )
+            .await;
+
+        match result {
+            Ok(()) => Ok(()),
+            Err(e) => Err(ConsumerStoreOffsetError::Client(e)),
+        }
+    }
```
Comment (wolf4ood) [inline src/consumer.rs:172]: You could make it a little bit shorter like this:

```suggestion
    pub async fn store_offset(&self, offset: u64) -> Result<(), ConsumerStoreOffsetError> {
        if let Some(name) = &self.name {
            self.internal
                .client
                .store_offset(name, self.internal.stream.as_str(), offset)
                .await
                .map(Ok)?
        } else {
            Err(ConsumerStoreOffsetError::NameMissing)
        }
    }
```

same for `query_offset`.

**Review by wolf4ood — APPROVED** [review]
LGTM, just a minor nit :)

**Comment by DanielePalaia** [comment]
@Gsantomaggio and @wolf4ood for the review! I also implemented the last suggestion as having more compact code is better!

**Pull request merged**